### PR TITLE
[IMP] mail: VoIP refactoring

### PR DESCRIPTION
**Pull request by oomsveta**
Adds useful fields to models needed for VoIP to work.

Enterprise: https://github.com/odoo/enterprise/pull/38339
Upgrade: https://github.com/odoo/upgrade/pull/5021

### CODE REVIEW

File: addons/mail/static/src/core/web/activity.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
                 </table>
             </div>
             <div t-if="activity.note" class="o-mail-Activity-note text-break" t-out="activity.note"/>
-            <div t-if="activity.mail_template_ids.length > 0" class="o-mail-Activity-mailTemplates">
+            <div t-if="activity.mail_template_ids?.length > 0" class="o-mail-Activity-mailTemplates">
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/activity.xml:42]: How comes this become an issue on this PR specifically?
Why not always have it as array?

File: addons/mail/static/src/core/common/persona_model.js [hunk 2]
```diff
@@ -15,6 +15,10 @@ export class Persona {
     localId;
     /** @type {number} */
     id;
+    /** @type {string} */
+    landlineNumber;
+    /** @type {string} */
+    mobileNumber;
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/persona_model.js:21]: I don't see why those are in community, they are never returned here?

**Review by alexkuhn — APPROVED** [review]

**Comment by seb-odoo** [comment]
runbot is red, comment in community never responded

File: addons/mail/static/src/core/common/persona_model.js (hunk 2 above)
Comment (oomsveta) [reply addons/mail/static/src/core/common/persona_model.js:58]: It avoids creating a patch for those fields that, even if not used on the client side, exist in Community

File: addons/mail/static/src/core/web/activity.xml (hunk 1 above)
Comment (oomsveta) [reply addons/mail/static/src/core/web/activity.xml:42]: I honestly have no idea, but without the optional chaining, opening the CRM app results in a crash 🤓

**Review by seb-odoo — APPROVED** [review]
@robodoo delegate+

**Comment by seb-odoo** [comment]
@robodoo r+

**Comment by oomsveta** [comment]
robodoo r+

**Comment by oomsveta** [comment]
robodoo r+

**Pull request closed without merging**